compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -685,6 +677,20 @@ impl DataFlowGraph {
     pub(crate) fn is_global(&self, value: ValueId) -> bool {
         matches!(self.values[value], Value::Global(_))
     }
+
+    pub(crate) fn get_instruction(&self, value: ValueId) -> Option<&Instruction> {
```

Comment:
I wonder if this should be the implementation of `Index` for `InstructionId`. What do you think? That way we don't have to remember to call `dfg.get_instruction(id)` instead of doing `dfg[id]`.

Reply:
`dfg[id]` wouldn't return an `Option` so may introduce panic if we sometimes look up non-instructions here.

Reply:
Yeah I thought the same thing but this would require some more changes.

From the PR description:
> In #6985 we only added a `Value::Global`, but globals support both numeric constants and array constants. This means that we can have overlapping instruction ids in the function graph and the globals graph. To avoid having to add placeholder instructions like we do for values, when checking for a constant array instruction we also check whether the value tied to that instruction is specified as a global. If the instruction is specified as global we look for the instruction in the globals graph.

This would most likely require some kind of `Instruction::Global` similar to `Value::Global`. If we feel it would be clearer to do this I can make the change.

Reply:
Oh, makes sense. This is fine then.